.github/workflows/semgrep.yml

```
@@ -0,0 +1,22 @@
+name: Semgrep
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/semgrep.yml
+  schedule:
+    - cron: '0 0 * * 0'
```

```suggestion
    - cron: '30 0 1,15 * *' # scheduled for 00:30 UTC on both the 1st and 15th of the month
```

Have Semgrep scans run periodically as well as reactively.